src/aof.c

```diff
@@ -759,6 +759,7 @@ int loadAppendOnlyFile(char *filename) {
         argv = zmalloc(sizeof(robj*)*argc);
         fakeClient->argc = argc;
         fakeClient->argv = argv;
+        fakeClient->argv_len = argc;
```

Comment:
This is indeed correct. But I don't understand how not setting this could cause a mem leak. The code in `freeClientArgv()` loops over `c->argc` to de-referrence  the `robj`s, so an invalid `argv_len` here sholdn't make a difference. What am I missing?

Reply:
Found it (thanks to @oranagra):
The specific command causing the mem leak (from the test suite) was:
```tcl
r xadd bigstream * entry1 $bigstr entry2 $bigstr
```
The `XADD` command implementation calls `rewriteClientCommandArgument()` to set the id arg inside the `argv` array. The implementation of `rewriteClientCommandArgument()` requires a correct `argv_len` to know if it needs to grow the `argv` array. Because `argv_len` was invalid, it actually shrank the array and updated `argc` to wrong (small) value. This lead to some arguments not being freed later on.